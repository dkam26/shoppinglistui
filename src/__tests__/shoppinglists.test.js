import React from 'react';
import { shallow, mount } from 'enzyme';
import {configure} from 'enzyme';
import Shoppinglists  from '../components/shoppinglists.component';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import toJson,{shallowToJson} from 'enzyme-to-json';
import LocalStorageMock from '../setupTests';
import Notifications, {notify} from 'react-notify-toast';
import ReactDOM from 'react-dom';
configure({ adapter: new Adapter() });
describe('shoopinglists component', () => {
    const props={
        addList:()=>{},
        fetchList :()=>{},
        editList:()=>{},
        getItems:()=>{},
        render:()=>{}

    }
    const wrapper =  shallow(<Shoppinglists />);
    it('should run', () => {
        expect(wrapper.exists(<div></div>)).toBe(true);
    });
    it('should render correctly', () => {
        const output = shallow( <Shoppinglists />);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
 it('should add shoppinglist',()=>{
        wrapper.find('#addshoppinglist').simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot();
       
        
    });
 it('shoppinglists should be a header',()=>{
    expect(wrapper.find('h3').length).toBe(1);
    expect(wrapper.find('h3').at(0).text()).toBe('Your Shoppinglists');
    });
    
it('render notifications',()=>{
        expect(wrapper.find('#alert').length).toBe(1);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Shoppinglists />, div);
    });
it('should ve the following methods',()=>{
     wrapper.instance().addList();
     wrapper.instance().fetchList ();
     wrapper.instance().editList();
     wrapper.instance().getItems();
     wrapper.instance().render();
    });
})