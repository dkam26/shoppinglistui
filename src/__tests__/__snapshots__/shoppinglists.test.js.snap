// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoopinglists component render notifications 1`] = `
<div>
  <Container>
    <Segment>
      <Menu
        secondary={true}
        style={
          Object {
            "backgroundColor": "black",
          }
        }
      >
        <MenuItem
          name="SHOPPINGLIST"
          style={
            Object {
              "color": "white",
            }
          }
        />
        <MenuMenu
          position="right"
        />
      </Menu>
    </Segment>
    <Segment>
      <Menu
        secondary={true}
      >
        <MenuMenu>
          <MenuItem>
            <Popup
              content="Add shoppinglist"
              id="addshoppinglist"
              on="hover"
              position="top left"
              trigger={
                <Button
                  as="button"
                  icon="add"
                  onClick={[Function]}
                />
              }
            />
          </MenuItem>
        </MenuMenu>
        <MenuMenu>
          <MenuItem>
            <h3
              style={
                Object {
                  "marginLeft": "308px",
                }
              }
            >
              Your Shoppinglists
            </h3>
          </MenuItem>
        </MenuMenu>
        <MenuMenu
          position="right"
        >
          <MenuItem>
            <Input
              icon="search"
              name="word"
              onChange={[Function]}
              onKeyUp={[Function]}
              placeholder="Search by name..."
              type="text"
            />
          </MenuItem>
        </MenuMenu>
      </Menu>
    </Segment>
    <p>
      No lists
    </p>
    <_class
      id="alert"
      options={Object {}}
    />
  </Container>
</div>
`;

exports[`shoopinglists component should add shoppinglist 1`] = `
<div>
  <Container>
    <Segment>
      <Menu
        secondary={true}
        style={
          Object {
            "backgroundColor": "black",
          }
        }
      >
        <MenuItem
          name="SHOPPINGLIST"
          style={
            Object {
              "color": "white",
            }
          }
        />
        <MenuMenu
          position="right"
        />
      </Menu>
    </Segment>
    <Segment>
      <Menu
        secondary={true}
      >
        <MenuMenu>
          <MenuItem>
            <Popup
              content="Add shoppinglist"
              id="addshoppinglist"
              on="hover"
              position="top left"
              trigger={
                <Button
                  as="button"
                  icon="add"
                  onClick={[Function]}
                />
              }
            />
          </MenuItem>
        </MenuMenu>
        <MenuMenu>
          <MenuItem>
            <h3
              style={
                Object {
                  "marginLeft": "308px",
                }
              }
            >
              Your Shoppinglists
            </h3>
          </MenuItem>
        </MenuMenu>
        <MenuMenu
          position="right"
        >
          <MenuItem>
            <Input
              icon="search"
              name="word"
              onChange={[Function]}
              onKeyUp={[Function]}
              placeholder="Search by name..."
              type="text"
            />
          </MenuItem>
        </MenuMenu>
      </Menu>
    </Segment>
    <p>
      No lists
    </p>
    <_class
      id="alert"
      options={Object {}}
    />
  </Container>
</div>
`;

exports[`shoopinglists component should render correctly 1`] = `
<div>
  <Container>
    <Segment>
      <Menu
        secondary={true}
        style={
          Object {
            "backgroundColor": "black",
          }
        }
      >
        <MenuItem
          name="SHOPPINGLIST"
          style={
            Object {
              "color": "white",
            }
          }
        />
        <MenuMenu
          position="right"
        />
      </Menu>
    </Segment>
    <Segment>
      <Menu
        secondary={true}
      >
        <MenuMenu>
          <MenuItem>
            <Popup
              content="Add shoppinglist"
              id="addshoppinglist"
              on="hover"
              position="top left"
              trigger={
                <Button
                  as="button"
                  icon="add"
                  onClick={[Function]}
                />
              }
            />
          </MenuItem>
        </MenuMenu>
        <MenuMenu>
          <MenuItem>
            <h3
              style={
                Object {
                  "marginLeft": "308px",
                }
              }
            >
              Your Shoppinglists
            </h3>
          </MenuItem>
        </MenuMenu>
        <MenuMenu
          position="right"
        >
          <MenuItem>
            <Input
              icon="search"
              name="word"
              onChange={[Function]}
              onKeyUp={[Function]}
              placeholder="Search by name..."
              type="text"
            />
          </MenuItem>
        </MenuMenu>
      </Menu>
    </Segment>
    <p>
      No lists
    </p>
    <_class
      id="alert"
      options={Object {}}
    />
  </Container>
</div>
`;
